lib/index.ts

```diff
@@ -0,0 +1,167 @@
+import * as socketio from 'socket.io';
+import * as rword from 'rword';
+import * as Constants from './Constants';
+import * as Types from './Types';
+import * as uuid from 'uuid';
+
+/**
+ * Important Notes:
+ *
+ * UserID is the Socket ID on connection.
+ * This socket id changes on page refreshes and reconnects from React service.
+ */
+
+// States
+
+const displayNames: Types.UserDisplayMap = {};
+const userRooms: Types.UserRoomMap = {};
+const rooms: Types.RoomMap = {};
+
+// Helper functions
+
+/**
+ * Logs message with non-error coloring. Also includes timestamp.
+ *
+ * @param info - message to log
+ */
+function logInfo(info: string): void {
+    console.log('\x1b[33m%s\x1b[0m', `[${new Date().toLocaleString()}]: ${info}`);
+}
+
+/**
+ * Logs message with error coloring. Also includes timestamp.
+ *
+ * @param info - error to log
+ */
+function logError(error: string): void {
+    console.log('\x1b[31m%s\x1b[0m', `[${new Date().toLocaleString()}]: ${error}`);
+}
+
+/**
+ * Returns random hex color
+ */
+function generateRandomColor(): string {
+    return (
+        '#' +
+        Math.random()
+            .toString(16)
+            .slice(2, 8)
+    );
+}
+
+// Socket Connection
+
+const io = socketio();
+io.on('connection', client => {
+    /**
+     * Received when client logs in as guest and sends display name and color
+     */
+    client.on(Constants.GET_DISPLAY_NAME, () => {
+        let displayName: string = (rword.rword.generate(2, { length: '2 - 6' }) as string[]).join('');
+        displayNames[client.id] = { userid: client.id, displayName, color: generateRandomColor() };
+        client.emit(Constants.DISPLAY_NAME, displayNames[client.id]);
+        io.emit(Constants.USERS, displayNames);
+    });
+
+    /**
+     * Received when client presses 'Create Room' button with relevant information.
+     */
+    client.on(Constants.CREATE_ROOM, (data: Types.CreateRoom) => {
+        let roomid = uuid.v1(); // Generate random time-based id
+        // In case id already exists (highly unlikely in theory)
```

Comment:
lol... hmm.......

Reply:
What does "time-based" mean?

Reply:
https://www.npmjs.com/package/uuid

Version 1: Timestamp.

Reply:
Oh if it's by timestamp, won't they be unique?